### DISCUSSION TITLE
Add favicon images to referrer?

### DISCUSSION THREAD

**Discussion by edent** [ibericode/koko-analytics#252] [Ideas]
Just experimenting with something like this.

![List of referrers with icons.](https://github.com/user-attachments/assets/f49a29c9-0a10-4520-8f52-61a179038fbf)

The icon comes from DuckDuckGo's free favicon service.  For example https://icons.duckduckgo.com/ip9/gigazine.net.ico

The format is `'https://icons.duckduckgo.com/ip9/' + t.displayUrl + '.ico'`

For example, see https://github.com/keepassxreboot/keepassxc/issues/2258

**Comment by dannyvankooten** [comment]
I like it, but personally don't see it becoming a core feature of Koko Analytics itself since it introduces a dependency (even though it's a soft one) on an external service plus it leaks some information about a website's referral info to this third party.

So probably not something that ends up being a default, but would be nice to support it for those that want it nonetheless. Nifty idea!

**Comment by dannyvankooten** [comment]
Version 1.4.0 added a new filter hook `koko_analytics_referrer_url_label` which should be relevant to this idea. 

_Note: while typing this I realize that the result of this filter hook is later escaped, so this isn't that useful for adding an `<img>` element before each referrer URL. We can work on that though, ie by escaping first and then running the filter._
